src/main/java/com/google/api/codegen/viewmodel/OptionalArrayMethodView.java

```diff
@@ -115,6 +116,16 @@ public boolean hasApiVersion() {
 
   public abstract List<HeaderRequestParamView> headerRequestParams();
 
+  /** All the configured value sets for all samples for this method. */
+  public abstract SampleValueSetsModel sampleValueSetsModel();
+
+  /**
+   * When this class is being used as a view model for a single sample, the one value set to be used
+   * in the sample.
+   */
+  @Nullable
+  public abstract SampleValueSet sampleValueSet();
```

Comment:
Codegen proto-generated types should not be put in the view model classes.

Reply:
OK, I created a `SampleValueSetsView` class and am using it in `SampleValueSetCollection` and elsewhere. I was going to do this sooner or later anyway because we'll need to pass the parsed representation of the initial values, as we do for the sample init fields.

    But I have to askwhy this blanket stricture if the proto-generated types are containers for the data we need that are already pre-defined and pre-populated? Is the concern about mutability? Would a deep copy work then?

    (As I said, this is moot right now, but I want to understand the thinking behind your comment.)